contract/src/main/kotlin/io/provenance/scope/loan/contracts/AppendLoanStatesContract.kt

```diff
@@ -23,22 +22,10 @@ open class AppendLoanStatesContract(
 
     @Function(invokedBy = PartyType.OWNER)
     @Record(LoanScopeFacts.servicingData)
-    open fun appendLoanStates(@Input(LoanScopeFacts.servicingData) newLoanStates: List<LoanStateMetadata>): ServicingData {
-        val updatedServicingData = ServicingData.newBuilder().mergeFrom(existingServicingData)
-        validateRequirements(VALID_INPUT) {
-            for (state in newLoanStates) {
-                // TODO: Improve check for duplicates (ID, effectiveTime, & checksum) & confirm if/which we want to silently ignore or raise violation
-                requireThat(
-                    state.id.isValid()              orError "Invalid id",
-                    state.effectiveTime.isValid()   orError "Invalid effective time",
-                    state.uri.isNotBlank()          orError "Invalid accrued interest",
-                    state.checksum.isValid()        orError "Invalid checksum"
-                )
-                if (existingServicingData.loanStateList.none { it.effectiveTime == state.effectiveTime }) {
-                    updatedServicingData.addLoanState(state)
-                }
+    open fun appendLoanStates(@Input(LoanScopeFacts.servicingData) newLoanStates: Collection<LoanStateMetadata>): ServicingData =
```

Comment:
Based on our Slack convo, I'd recommend updating the `@Input` name here to be `newLoanStates` or just `loanStates` so it's not confusing to the caller.

Reply:
Done in #21